Closes #1

Neonchick opened the pull request:
Добовение списка

Commit:
Closes #1

Добовение списка

HimmelSpark commented on Recipista/.idea/codeStyles/Project.xml:
```diff
@@ -0,0 +1,134 @@
+<component name="ProjectCodeStyleConfiguration">
```
@Neonchick 
Уверен, что файлы из .idea должны быть в репозитории?

HimmelSpark commented on Recipista/app/build.gradle:
```diff
@@ -25,8 +25,9 @@ android {
     implementation 'androidx.appcompat:appcompat:1.2.0'
-    implementation 'androidx.constraintlayout:constraintlayout:2.0.2'
-    testImplementation 'junit:junit:4.12'
+    implementation 'androidx.constraintlayout:constraintlayout:2.0.4'
```
Версии надо вынести в глобальный build.gradle
В глобальном файле добавляешь следующее:
```
configure(subprojects) { subProject ->
    ext {
        daggerVersion = '2.17'
    }
}
```
А в этом build.gradle импортишь так:
```
implementation group: 'com.google.dagger', name:'dagger', version: daggerVersion
```

HimmelSpark commented on Recipista/app/src/main/java/ru/mail/hse/recipista/IngredientAdapter.java:
```diff
@@ -0,0 +1,41 @@
+package ru.mail.hse.recipista;
```
Запихни в пакет adapter

HimmelSpark commented on Recipista/app/src/main/java/ru/mail/hse/recipista/IngredientAdapter.java:
```diff
@@ -0,0 +1,41 @@
+    @NonNull
+    @Override
+    public IngredientViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        // Получаем инфлейтер и создаем нужный layout
```
Комменты не нужны. Только, если что-то нестандартное и неочевидное либо пометка на будущее

HimmelSpark commented on Recipista/app/src/main/java/ru/mail/hse/recipista/IngredientAdapter.java:
```diff
@@ -0,0 +1,41 @@
+        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
+        View layout = inflater.inflate(R.layout.list_element, parent, false);
+
+        // Создаем ViewHolder
```
См выше

HimmelSpark commented on Recipista/app/src/main/java/ru/mail/hse/recipista/IngredientAdapter.java:
```diff
@@ -0,0 +1,41 @@
+
+class IngredientAdapter extends RecyclerView.Adapter<IngredientViewHolder> {
+
+    List<String> ingredients;
```
private

HimmelSpark commented on Recipista/app/src/main/java/ru/mail/hse/recipista/IngredientFragment.java:
```diff
@@ -0,0 +1,38 @@
+package ru.mail.hse.recipista;
```
Запихни в пакет fragment

HimmelSpark commented on Recipista/app/src/main/java/ru/mail/hse/recipista/IngredientFragment.java:
```diff
@@ -0,0 +1,38 @@
+import androidx.recyclerview.widget.RecyclerView;
+
+public class IngredientFragment extends Fragment {
+    RecyclerView recycler = null;
```
private

HimmelSpark commented on Recipista/app/src/main/java/ru/mail/hse/recipista/IngredientViewHolder.java:
```diff
@@ -0,0 +1,20 @@
+
+class IngredientViewHolder extends RecyclerView.ViewHolder {
+
+    TextView ingredientTV = itemView.findViewById(R.id.ingredient_tv);
```
private

HimmelSpark commented on Recipista/app/src/main/java/ru/mail/hse/recipista/IngredientViewHolder.java:
```diff
@@ -0,0 +1,20 @@
+package ru.mail.hse.recipista;
```
В пакет viewholder

HimmelSpark commented on Recipista/app/src/main/java/ru/mail/hse/recipista/IngredinentRepository.java:
```diff
@@ -0,0 +1,64 @@
+package ru.mail.hse.recipista;
```
В пакет repository

HimmelSpark commented on Recipista/app/src/main/java/ru/mail/hse/recipista/IngredinentRepository.java:
```diff
@@ -0,0 +1,64 @@
+    // Объекты для реализации хардкорного синглтона в java
+    private static volatile IngredinentRepository mInstance;
+
+    public static IngredinentRepository getInstance() {
```
Если сделать статическую инициализацию списка, то можно не городить синглтон

HimmelSpark commented on Recipista/app/src/main/java/ru/mail/hse/recipista/IngredinentRepository.java:
```diff
@@ -0,0 +1,64 @@
+import java.util.Random;
+
+public class IngredinentRepository {
+    // Константы
```
убери комменты

HimmelSpark commented on Recipista/app/src/main/java/ru/mail/hse/recipista/IngredinentRepository.java:
```diff
@@ -0,0 +1,64 @@
+
+        return data;
+    }
+}
```
добавь пустую строку в конец

HimmelSpark commented on Recipista/app/src/main/java/ru/mail/hse/recipista/MainActivity.java:
```diff
@@ -6,9 +6,15 @@
 
 public class MainActivity extends AppCompatActivity {
 
+    final String TAG_LIST = "LIST";
```
Возможно в будущем будем извлекать для чего-то этот ресайклер по этому тегу
Давай сделаем класс со статическими тегами. Мне кажется такое надо выносить в отдельное место.
Что думаешь?

HimmelSpark replied:
private)

HimmelSpark commented on Recipista/app/src/main/java/ru/mail/hse/recipista/MainActivity.java:
```diff
@@ -6,9 +6,15 @@
+                .add(R.id.main_layout, new IngredientFragment(), TAG_LIST)
+                .commit();
     }
 }
```
add newline

HimmelSpark replied:
newline

HimmelSpark commented on Recipista/app/src/main/res/layout/activity_main.xml:
```diff
@@ -1,18 +1,6 @@
+    xmlns:android = "http://schemas.android.com/apk/res/android"
+    android:layout_width = "match_parent"
+    android:layout_height = "match_parent"
+    android:id="@+id/main_layout"/>
```
add newline

HimmelSpark commented on Recipista/app/src/main/res/layout/list_element.xml:
```diff
@@ -0,0 +1,17 @@
+        app:layout_constraintTop_toTopOf="parent"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintLeft_toLeftOf="parent"/>
+</androidx.constraintlayout.widget.ConstraintLayout>
```
newline

HimmelSpark commented on Recipista/app/src/main/res/layout/list_fragment.xml:
```diff
@@ -0,0 +1,10 @@
+        android:id = "@+id/recycler"
+        android:layout_width = "match_parent"
+        android:layout_height = "match_parent"/>
+</FrameLayout>
```
newline

HimmelSpark commented on Recipista/app/src/main/res/values/dimens.xml:
```diff
@@ -0,0 +1,5 @@
+<resources>
+    <dimen name="standart_margin">20dp</dimen>
+    <dimen name="min_listElement_heigth">40dp</dimen>
+</resources>
```
newline

HimmelSpark commented on Recipista/.idea/gradle.xml:
```diff
@@ -0,0 +1,21 @@
+      </GradleProjectSettings>
+    </option>
+  </component>
+</project>
```
newline

HimmelSpark commented on Recipista/.idea/codeStyles/codeStyleConfig.xml:
```diff
@@ -0,0 +1,5 @@
+  <state>
+    <option name="PREFERRED_PROJECT_CODE_STYLE" value="Default" />
+  </state>
+</component>
```
newline

HimmelSpark commented on Recipista/.idea/codeStyles/Project.xml:
```diff
@@ -0,0 +1,134 @@
+      </arrangement>
+    </codeStyleSettings>
+  </code_scheme>
+</component>
```
newline

HimmelSpark commented on Recipista/.idea/jarRepositories.xml:
```diff
@@ -0,0 +1,25 @@
+      <option name="url" value="https://dl.google.com/dl/android/maven2/" />
+    </remote-repository>
+  </component>
+</project>
```
newline

HimmelSpark commented on Recipista/.idea/misc.xml:
```diff
@@ -0,0 +1,6 @@
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_7" project-jdk-name="1.8" project-jdk-type="JavaSDK">
+    <output url="file://$PROJECT_DIR$/build/classes" />
+  </component>
+</project>
```
newline

HimmelSpark commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Сделать ресайклер](https://trello.com/c/DHhH34k3/1-%D1%81%D0%B4%D0%B5%D0%BB%D0%B0%D1%82%D1%8C-%D1%80%D0%B5%D1%81%D0%B0%D0%B9%D0%BA%D0%BB%D0%B5%D1%80)

HimmelSpark commented:
@Neonchick 
Подтяни в свою ветку актуальный девелоп

Commit:
Merge branch 'develop' into feature-simple_recycle_view

HimmelSpark committed:
closes #1
 - вынес градловые версии в parent project

HimmelSpark committed:
Merge remote-tracking branch 'origin/feature-simple_recycle_view' into feature-simple_recycle_view

Neonchick reviewed (commented):
Теперь понял

Commit:
Closes #1

- Классы перенесены в пакеты
- Добаленны модификаторы private
- Добавлен класс тегов
- Убран синглтон у статического списка

HimmelSpark committed:
closes #1
 - правка в IngredientRepository
